poly/src/evaluations/multivariate/multilinear/dense.rs

```diff
@@ -0,0 +1,400 @@
+//! multilinear polynomial represented in dense evaluation form.
+
+use crate::evaluations::multivariate::multilinear::{swap_bits, MultilinearExtension};
+use ark_ff::{Field, Zero};
+use ark_serialize::{CanonicalDeserialize, CanonicalSerialize, Read, SerializationError, Write};
+use ark_std::fmt;
+use ark_std::fmt::Formatter;
+use ark_std::ops::{Add, AddAssign, Index, Neg, Sub, SubAssign};
+use ark_std::slice::{Iter, IterMut};
+use ark_std::vec::Vec;
+use rand::Rng;
+#[cfg(feature = "parallel")]
+use rayon::prelude::*;
+
+/// Stores a multilinear polynomial in dense evaluation form.
+#[derive(Clone, PartialEq, Eq, Hash, Default, CanonicalSerialize, CanonicalDeserialize)]
+pub struct DenseMultilinearExtension<F: Field> {
+    /// The evaluation over {0,1}^n
```

Comment:
What is `n` here?

Reply:
typo, should be {0,1}^`num_vars`. I'll fix that